src/libraries/System.Threading.RateLimiting/src/System/Threading/RateLimiting/ChainedRateLimiter.cs

```diff
@@ -0,0 +1,191 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Threading.RateLimiting
+{
+    /// <summary>
+    /// Acquires leases from rate limiters in the order given. If a lease fails to be acquired (throwing or IsAcquired == false)
+    /// then the already acquired leases are disposed in reverse order and the failing lease is returned or the exception is thrown to user code.
+    /// </summary>
+    internal sealed partial class ChainedRateLimiter : RateLimiter
+    {
+        private readonly RateLimiter[] _limiters;
+        private bool _disposed;
+
+        public ChainedRateLimiter(RateLimiter[] limiters)
+        {
+            _limiters = limiters;
```

Comment:
This is storing the array the caller provided? There's no indication of ownership transfer; something should be making a copy, or else the caller could change the array after constructing the limiter and those changes would have undefined effects on this implementation.

Reply:
Good remark!
I have added the clone of the provided array.
I have also added it to the existing `ChainedPartitionedRateLimiter` implementation.